### DISCUSSION TITLE
Generating subdomains from the element list?

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#18444] [Q&A General] [answered]
Hi guys, 

is it possible to generate subdomains by element list?

from the list under `meshgenerators`, it doesn't seem like there is one that can do what I want

I could do that by using `FileMeshGenerator` to import pre-defined subdomains from .inp file (Abaqus).

---------------------------------------------------------------------------------------------------

my ultimate goal is to first, create cohesive surfaces from the subdomains (red/large blocks) using `BreakMeshByBlockGenerator `then impose the mining-step subdomains into the main mesh (black/small blocks)

so generating subdomains from the element list will help in the later part.

![image](https://user-images.githubusercontent.com/79562439/126920795-d2878b66-c97e-43cf-8833-64eb3869c623.png)

@jiangwen84 @arovinelli, gents, I think this approach can help with my CZM problem, what do you think?

Kind regards,
Traiwit

**Comment by GiudGiud** [comment]
Hello

There is not. You can use https://mooseframework.inl.gov/source/meshgenerators/ParsedSubdomainMeshGenerator.html to sort of define the block based on the location of the centroid of the element, which could work here if you convert each block shape into a combinatorial geometry expression. 
If you want to work with a list, I think modifying the .inp file, or specifying subblocks in abacus is the way to go here. 

Guillaume

**Comment by arovinelli** [comment, marked as answer]
[ElementSubdomainIDGenerator](https://mooseframework.inl.gov/source/meshgenerators/ElementSubdomainIDGenerator.html)
is similar to what you need.
The bad part is you have to give all elements a new block id.
The element list is implied to be 0,1,2,...,N
The example below will give to elements  0,1,2,3,4 block id 0 2 3 1 2

```
  [./subdomains]
    type = ElementSubdomainIDGenerator
    input = gmg
    subdomain_ids = '0 2 3 1 2'
  []
```
